### PR TITLE
feat: add e2e unittest framework

### PR DESCRIPTION
**Pull request by dongjiang1989**
Add e2e unittest framework and add One case for CRD Install

### CODE REVIEW

**Comment by dongjiang1989** [comment]
@peter-wangxu  PTAL

**Comment by peter-wangxu** [comment]
Do you plan adding a example test case other than install CRD?

**Comment by dongjiang1989** [comment]
> Do you plan adding a example test case other than install CRD?

Yes. I will continue to add e2e cases.

**Comment by dongjiang1989** [comment]
@peter-wangxu  It's need to update token?
```
[2024-01-08T11:40:03.070Z] ['info'] Pinging Codecov: https://codecov.io/upload/v4?package=github-action-2.1.0-uploader-0.7.1&token=*******&branch=e2e-framework&build=7446893003&build_url=https%3A%2F%2Fgithub.com%2Falibaba%2Fopen-local%2Factions%2Fruns%2F7446893003%2Fjob%2F20258074232&commit=d8411a0ab6dfa9eb3ef6c75eca8cbc54579387b5&job=Build&pr=251&service=github-actions&slug=alibaba%2Fopen-local&name=&tag=&flags=unittests&parent=
[2024-01-08T11:40:03.374Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2/dist/codecov' failed with exit code 255
```

File: test/e2e/agent_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+					},
+				},
+			},
+		})
+
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
```
Comment (dongjiang1989) [inline test/e2e/agent_test.go:73]: @peter-wangxu  this is example e2e case.

**Comment by dongjiang1989** [comment]
> Did you pass local testing? If yes i am ok to merge this now.

Yes. Please merge it.  I will continue to add e2e cases.

**Pull request merged**